internal/internal_event_handlers_test.go

```diff
@@ -402,3 +406,322 @@ func TestProcessQuery_KnownQueryTypes(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, "[\"__open_sessions\",\"__query_types\",\"__stack_trace\",\"a\"]\n", string(result))
 }
+
+func TestWorkflowExecutionEventHandler_ProcessEvent_WorkflowExecutionStarted(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		testRegistry := newRegistry()
+		testRegistry.RegisterWorkflowWithOptions(func(ctx Context) error { return nil }, RegisterWorkflowOptions{Name: "test"})
+
+		weh := testWorkflowExecutionEventHandler(t, testRegistry)
+
+		testHeaderStruct := &s.Header{
+			Fields: map[string][]byte{
+				"test": []byte("test"),
+			},
+		}
+		testInput := []byte("testInput")
+		event := &s.HistoryEvent{
+			EventType: common.EventTypePtr(s.EventTypeWorkflowExecutionStarted),
+			WorkflowExecutionStartedEventAttributes: &s.WorkflowExecutionStartedEventAttributes{
+				WorkflowType: &s.WorkflowType{
+					Name: common.StringPtr("test"),
+				},
+				Input:  testInput,
+				Header: testHeaderStruct,
+			},
+		}
+
+		err := weh.ProcessEvent(event, false, false)
+		assert.NoError(t, err)
```

Comment:
If at all possible to check something other than no error, I think it would be good. But sometimes it's not